### PR TITLE
Update Jira references from HUDSON to JENKINS

### PR DESCRIPTION
**Pull request by daniel-beck**
### Proposed changelog entries

Internal only

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by basil — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by res0nance — APPROVED** [review]

File: core/src/main/java/hudson/Functions.java [hunk 1]
```diff
@@ -1947,7 +1947,7 @@ public boolean hyperlinkMatchesCurrentPage(String href) throws UnsupportedEncodi
      * Gets all the {@link PageDecorator}s.
      */
     public static List<PageDecorator> getPageDecorators() {
-        // this method may be called to render start up errors, at which point Hudson doesn't exist yet. see HUDSON-3608 
+        // this method may be called to render start up errors, at which point Hudson doesn't exist yet. see JENKINS-3608
```
Comment (StefanSpieker) [inline core/src/main/java/hudson/Functions.java:1950]:
```suggestion
        // this method may be called to render start up errors, at which point Jenkins doesn't exist yet. see JENKINS-3608
```

File: core/src/test/java/hudson/scheduler/CronTabDayOfWeekLocaleTest.java [hunk 2]
```diff
@@ -50,8 +49,8 @@ public CronTabDayOfWeekLocaleTest(Locale locale) {
      * HUDSON-8656.
      */
     @Test
-    @Url("http://issues.hudson-ci.org/browse/HUDSON-8656")
-    public void hudson8658() throws Exception {
+    @Issue("HUDSON-8656") // This is _not_ JENKINS-8656
+    public void hudson8656() throws Exception {
```
Comment (StefanSpieker) [inline core/src/test/java/hudson/scheduler/CronTabDayOfWeekLocaleTest.java:53]: Why do you rename the test?

**Review by StefanSpieker — APPROVED** [review]

File: core/src/test/java/hudson/scheduler/CronTabDayOfWeekLocaleTest.java (hunk 2 above)
Comment (daniel-beck) [reply core/src/test/java/hudson/scheduler/CronTabDayOfWeekLocaleTest.java:53]: Apparent typo in test name. Per the comment this is straight from HUDSON-8656 after the split (and no HUDSON-8658 is archived for us to check).

> Screenshot

File: core/src/main/java/hudson/Functions.java (hunk 1 above)
Comment (oleg-nenashev) [reply core/src/main/java/hudson/Functions.java:1950]: Can be handled separately as a bulk Javadoc cleanup

**Comment by oleg-nenashev** [comment]
I will go ahead and merge it so that it gets into the weekly

**Pull request merged**